lib/Syntax/SyntaxNodes.cpp.gyb

```diff
@@ -99,3 +99,8 @@ ${node.name} ${node.name}::with${child.name}(
 
 %   end
 % end
+
+//===----------------------- C API ----------------------------------------===//
+const char* swiftparse_node_declaration_hash() {
```

Comment:
It's not a good idea to spread around the C API implementation in different libraries. The C API implementation should stay in `tools/libSwiftSyntaxParser`, so either gyb-generate the string in `libSwiftSyntaxParser` or generate it in another library and have the C function call into that library.

Also `libSwiftSyntaxParser` is not depending on `swiftSyntax` library, it only depends on `swiftParse`.

Reply:
> Also `libSwiftSyntaxParser` is not depending on `swiftSyntax` library, it only depends on `swiftParse`.

Ignore that, `swiftParse` depends on `swiftSyntax` so it's ok to put it there.

Reply:
Done with the change. I'd like to keep the parser lib gyb-free so I used the second option.